tests/topotests/bgp_vpnv4_per_nexthop_label/test_bgp_vpnv4_per_nexthop_label.py

```diff
@@ -0,0 +1,394 @@
+    bgp_vpnv4_table_check_all(tgen.gears["r2"], labelList)
+
+    # Check MPLS labels received on r1
+    mpls_table_check(tgen.gears["r1"], labelList)
+
+
+def test_flapping_bgp_vrf_down():
+    """
+    Turn down a remote BGP session
+    """
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+    logger.info("Unpeering BGP on r11")
+    dump = tgen.gears["r11"].vtysh_cmd(
+        "configure terminal\nrouter bgp 65500\nno neighbor 192.0.2.1\n",
+        isjson=False,
+    )
+
+    def _bgp_failed(router):
+        output = json.loads(
+            router.vtysh_cmd("show ip bgp vrf vrf1 neighbor 192.0.2.11 json")
+        )
+        expected = {
+            "192.0.2.11": {
+                "bgpState": "Active",
+            }
+        }
+        return topotest.json_cmp(output, expected)
+
+    # Check r11 peering is not established
+    test_func = functools.partial(_bgp_failed, tgen.gears["r1"])
+    success, result = topotest.run_and_expect(test_func, None, count=10, wait=0.5)
+    assert success, "r1, peering with r11 is not failed"
+
+    # Check BGP updated received on r2 are not from r11
+    logger.info("Checking BGP VPNv4 labels on r2")
+    group1 = ["172.31.0.11/32", "172.31.0.20/32", "172.31.0.111/32"]
+    for entry in group1:
+        dump = tgen.gears["r2"].vtysh_cmd(
+            "show bgp ipv4 vpn {} json".format(entry), isjson=True
+        )
+        for rd in dump:
+            assert False, "r2, {}, route distinguisher {} present".format(entry, rd)
+
+    mpls_table_check(tgen.gears["r1"], blacklist=["192.0.2.11"])
+
+
+def test_flapping_bgp_vrf_up():
+    """
+    Turn up a remote BGP session
+    """
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+    logger.info("Peering BGP on r11")
+    dump = tgen.gears["r11"].vtysh_cmd(
+        "configure terminal\nrouter bgp 65500\nneighbor 192.0.2.1 remote-as 65500\n",
+        isjson=False,
+    )
+
+    def _bgp_established(router):
+        output = json.loads(
+            router.vtysh_cmd("show ip bgp vrf vrf1 neighbor 192.0.2.11 json")
+        )
+        expected = {
+            "192.0.2.11": {
+                "bgpState": "Established",
+            }
+        }
+        return topotest.json_cmp(output, expected)
+
+    # Check r11 peering is established
+    test_func = functools.partial(_bgp_established, tgen.gears["r1"])
+    success, result = topotest.run_and_expect(test_func, None, count=10, wait=0.5)
+    assert success, "r1, peering with r11 is not established"
+    bgp_vpnv4_table_check_all(tgen.gears["r2"])
+
+
+def test_dump_bgp_label_nexthop():
+    """
+    Display label nexthop list
+    """
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    logger.info("Dumping label nexthop table")
+    dump = tgen.gears["r1"].vtysh_cmd(
+        "show bgp vrf vrf1 label-nexthop detail", isjson=False
+    )
```

Comment:
You are not checking the dump

Reply:
as said, a comment is added to say it is only used for diagnostic